File: app/src/main/java/com/odnovolov/randomuser/screen/userlist/UserListView.java
package com.odnovolov.randomuser.screen.userlist;

import com.odnovolov.randomuser.model.User;

import java.util.List;

import moxy.MvpView;
import moxy.viewstate.strategy.AddToEndSingleTagStrategy;
import moxy.viewstate.strategy.StateStrategyType;

public interface UserListView extends MvpView {
    @StateStrategyType(value = AddToEndSingleTagStrategy.class, tag = "status")
    void showUsers(List<User> users);

    @StateStrategyType(value = AddToEndSingleTagStrategy.class, tag = "status")
    void showProgress();

    @StateStrategyType(value = AddToEndSingleTagStrategy.class, tag = "status")
    void showError();
}

File: app/src/main/java/com/odnovolov/randomuser/screen/userlist/UserListView.kt
package com.odnovolov.randomuser.screen.userlist

import com.odnovolov.randomuser.model.User
import moxy.MvpView
import moxy.viewstate.strategy.AddToEndSingleTagStrategy
import moxy.viewstate.strategy.StateStrategyType
import moxy.viewstate.strategy.alias.OneExecution

interface UserListView : MvpView {
    @StateStrategyType(value = AddToEndSingleTagStrategy::class, tag = "status")
    fun showUsers(users: List<User>)

    @StateStrategyType(value = AddToEndSingleTagStrategy::class, tag = "status")
    fun showProgress()

    @StateStrategyType(value = AddToEndSingleTagStrategy::class, tag = "status")
    fun showError()
}